### PR TITLE
Bug/condo/sberdoma 383/tickets list and xls export

### PR DESCRIPTION
**Pull request by dkoviazin**
1. Page size now can be changed
2. Xls export will export all tickets (limit 100 - because of server-side limit to 1000 requests in total ) sorted and filtered

### CODE REVIEW

File: apps/condo/domains/ticket/utils/helpers.spec.ts [hunk 1]
```diff
@@ -405,6 +407,19 @@ describe('Helpers', () => {
...
 
+        describe('getPageSizeFromQuery', () => {
+            describe('extract page size based on pagesize string from query object if pagesize presnts or set it to default', () => {
+
+                it('is valid when setted to value not from enum', () => {
+                    expect(getPageSizeFromQuery({ pagesize: '16' })).toStrictEqual(20)
+                })
+
+                it('is valid when not defined', () => {
+                    expect(getPageSizeFromQuery({ offset: undefined })).toStrictEqual(TICKET_PAGE_SIZE)
```
Comment (AntonAL) [inline apps/condo/domains/ticket/utils/helpers.spec.ts:418]: I don't see a working with `offset` param in `getPageSizeFromQuery`.
Maybe, `pagesize`?

File: apps/condo/domains/ticket/utils/helpers.spec.ts [hunk 2]
```diff
@@ -405,6 +407,19 @@ describe('Helpers', () => {
 
         })
 
+        describe('getPageSizeFromQuery', () => {
+            describe('extract page size based on pagesize string from query object if pagesize presnts or set it to default', () => {
+
+                it('is valid when setted to value not from enum', () => {
```
Comment (AntonAL) [inline apps/condo/domains/ticket/utils/helpers.spec.ts:413]:
```suggestion
                it('returns nearest value to provided `pagesize` query param, presented in enum', () => {
```

File: apps/condo/domains/ticket/utils/helpers.spec.ts [hunk 3]
```diff
@@ -405,6 +407,19 @@ describe('Helpers', () => {
...
+                    expect(getPageSizeFromQuery({ pagesize: '16' })).toStrictEqual(20)
+                })
+
+                it('is valid when not defined', () => {
```
Comment (AntonAL) [inline apps/condo/domains/ticket/utils/helpers.spec.ts:417]:
```suggestion
                it('returns default page size if no `pagesize` query param is provided', () => {
```

File: apps/condo/domains/ticket/utils/helpers.ts [hunk 4]
```diff
@@ -314,9 +314,22 @@ export const getSortStringFromQuery = (query: ParsedUrlQuery): Array<string> =>
...
+    if (POSSIBLE_PAGE_SIZE.indexOf(queryValue) !== -1) {
+        return queryValue
+    }
+    const nearest = POSSIBLE_PAGE_SIZE
+        .map(validPageSize => ({ validPageSize, difference: Math.abs(queryValue - validPageSize) }))
+        .sort((a, b) => a.difference - b.difference)[0].validPageSize
+    return nearest
+}
 
 export const getPaginationFromQuery = (query: ParsedUrlQuery): number => {
```
Comment (AntonAL) [inline apps/condo/domains/ticket/utils/helpers.ts:331]: If it returns a page index, maybe it's better to rename it as `getPageIndexFromQuery`?

**Review by Dimitreee — APPROVED** [review]

File: apps/condo/domains/ticket/utils/helpers.spec.ts (hunk 1 above)
Comment (dkoviazin) [reply apps/condo/domains/ticket/utils/helpers.spec.ts:418]: Done

File: apps/condo/domains/ticket/utils/helpers.spec.ts (hunk 2 above)
Comment (dkoviazin) [reply apps/condo/domains/ticket/utils/helpers.spec.ts:413]: Done

File: apps/condo/domains/ticket/utils/helpers.spec.ts (hunk 3 above)
Comment (dkoviazin) [reply apps/condo/domains/ticket/utils/helpers.spec.ts:417]: Done

File: apps/condo/domains/ticket/utils/helpers.ts (hunk 4 above)
Comment (dkoviazin) [reply apps/condo/domains/ticket/utils/helpers.ts:331]: Done

File: apps/condo/pages/ticket/index.tsx [hunk 5]
```diff
@@ -151,7 +178,14 @@ const TicketsPage = () => {
                                         />
                                     </Col>
                                     <Col span={6} push={1}>
-                                        <Button type={'inlineLink'} icon={<DatabaseFilled />} onClick={generateExcelData}>{ExportAsExcel}</Button>
+                                        <Button
```
[the remark this answers is not in the document]
Comment (dkoviazin) [reply apps/condo/pages/ticket/index.tsx:181]: Done

**Review by Dimitreee — APPROVED** [review]

**Pull request merged**